codec_packetizers/h264/h264_depacketizer.c

```diff
@@ -93,6 +96,78 @@ static H264Result_t DepacketizeFragmentationUnitPacket( H264DepacketizerContext_
 
 /*-----------------------------------------------------------*/
 
+static H264Result_t DepacketizeAggregationPacket( H264DepacketizerContext_t * pCtx,
+                                                  Nalu_t * pNalu )
+{
+    uint8_t * pCurPacketData;
+    size_t curPacketLength, naluLength;
+    H264Result_t result = H264_RESULT_OK;
+
+    pCurPacketData = pCtx->pPacketsArray[ pCtx->tailIndex ].pPacketData;
+    curPacketLength = pCtx->pPacketsArray[ pCtx->tailIndex ].packetDataLength;
+
+    /* We are just starting to parse a STAP-A packet. Skip the STAP-A header. */
+    if( pCtx->curPacketIndex == 0 )
+    {
+        pCtx->curPacketIndex += STAP_A_HEADER_SIZE;
+    }
+
+    /* Is there enough data left in the packet to read the next NALU size? */
+    if( ( pCtx->curPacketIndex + STAP_A_NALU_SIZE ) <= curPacketLength )
+    {
+        /* Read NALU length. */
+        naluLength = pCurPacketData[ pCtx->curPacketIndex ];
+        naluLength = ( naluLength << 8 ) |
+                     ( pCurPacketData[ pCtx->curPacketIndex + 1 ] );
+
+        pCtx->curPacketIndex += STAP_A_NALU_SIZE;
+
+        /* Is there enough data left in the packet to read the next NALU? */
+        if( ( naluLength > 0 ) &&
```

Comment:
IMHO, we should take this as pass case instead of malformed when nalu length is equal to 0.